service/src/main/java/com/codeforcommunity/processor/ReservationProcessorImpl.java

```diff
@@ -0,0 +1,68 @@
+package com.codeforcommunity.processor;
+
+import static org.jooq.generated.Tables.RESERVATIONS;
+
+import com.codeforcommunity.api.IReservationProcessor;
+import com.codeforcommunity.auth.JWTData;
+import com.codeforcommunity.dto.reservation.*;
+import org.jooq.DSLContext;
+import org.jooq.generated.tables.records.ReservationsRecord;
+
+import java.sql.Timestamp;
+import java.util.Optional;
+
+public class ReservationProcessorImpl implements IReservationProcessor {
+
+    private final DSLContext db;
+
+    public ReservationProcessorImpl(DSLContext db) {
+        this.db = db;
+    }
+
+
+    @Override
+    public void makeReservation(JWTData userData, MakeReservationRequest makeReservationRequest) {
```

Comment:
Needs checks for
1. If the userId exists
2. If the user has the privilege level of executing this action (May not be relevant)
2. If the blockId exists
3. If the teamId exists
4. If the given user is on the given team

A lot of these checks will be repeated for all of these routes so feel free to try to pull that into a helper function if it comes out cleanly.